packages/config/eslint-preset.js

```diff
@@ -1,40 +1,148 @@
-/** @type {import("eslint").Linter.Config} */
+const path = require("node:path");
+
+/**
+ * @type {import("eslint").Linter.Config}
+ * */
 module.exports = {
-  root: true,
-  extends: ["plugin:prettier/recommended", "turbo"],
-  parser: "@typescript-eslint/parser",
-  parserOptions: {
-    tsconfigRootDir: __dirname,
-    sourceType: "module",
-    project: ["./packages/*/tsconfig.json"],
-  },
-  plugins: ["@typescript-eslint"],
-  overrides: [
-    {
-      files: ["*.ts", "*.tsx"],
-      extends: ["plugin:@typescript-eslint/recommended", "plugin:@typescript-eslint/eslint-recommended"],
-      plugins: ["@typescript-eslint"],
-      parser: "@typescript-eslint/parser",
-      rules: {
-        "@typescript-eslint/no-empty-interface": 0,
-      },
+    env: {
+        browser: true,
+        es6: true,
+        node: true,
     },
-    {
-      files: ["packages/db/src/migrations/*"],
-      rules: {
-        "@typescript-eslint/no-explicit-any": "off",
-      },
+
+    extends: [
+        "eslint:recommended",
+        "plugin:@typescript-eslint/recommended",
+        "plugin:@typescript-eslint/stylistic",
+        "plugin:perfectionist/recommended-natural",
+        "plugin:react/recommended",
+        "plugin:react-hooks/recommended",
+        "plugin:jsx-a11y/recommended",
+        "turbo",
+        "prettier",
+    ],
+
+    globals: {
+        React: "writable",
     },
-  ],
-  rules: {
-    "@typescript-eslint/no-unused-vars": [
-      "warn",
-      {
-        argsIgnorePattern: "^_",
-        varsIgnorePattern: "^_",
-        caughtErrorsIgnorePattern: "^_",
-      },
+
+    ignorePatterns: ["*.stories.tsx", "dist/", "node_modules/", "apps/docs/"],
+
+    overrides: [
+        {
+            files: ["packages/db/src/migrations/*"],
+            rules: {
+                "@typescript-eslint/no-explicit-any": "off",
+            },
+        },
     ],
-  },
-  ignorePatterns: ["**/*.stories.tsx", "**/dist", "node_modules", "apps/docs"],
-}
+
+    parser: "@typescript-eslint/parser",
+    parserOptions: {
+        project: [path.resolve(__dirname, "tsconfig.json"), "./packages/*/tsconfig.json"],
+        sourceType: "module",
+        tsconfigRootDir: __dirname,
+    },
+
+    plugins: ["@typescript-eslint"],
+
+    root: true,
+
+    rules: {
```

Comment:
Could you explain the rules added, and the value of each of them? I fear we might be adding a lot of restrictions for little gain